### PR TITLE
Improve HTML tag stripping and update documentation

### PR DESCRIPTION
**Pull request by speedyk-005**
### Objective

Doctype declarations, comments, processing instructions, script and style blocks were leaking through the HTML cleaner. The old regex also had a backtracking problem with void elements like `` and `<embed>`. And it mangled math expressions like `x < 5 and y > 3`.

### Changes

- **New branches** for HTML comments, doctype declarations, and processing instructions. These now get stripped properly.
- **Container stripping** now handles 9 elements (`script`, `style`, `iframe`, `object`, `code`, `noscript`, `svg`, `canvas`, `template`). Tag plus content removed.
- **Void element fix**. Removed `img` and `embed` from the content-stripping group. These elements have no closing tag, so the old regex would search the whole document looking for `</img>` that never arrives. The result was about **37x slower** on documents with many `` tags.
- **`[a-zA-Z]` guard**. The last tag-stripping branch now requires a letter after `<` or `</`. So `x < 5 and y > 3` stays intact.
- README now describes the cleaner as a "lightweight preprocessor, not a full HTML parser".

### Verification

- 22 edge case tests pass (math expressions, normal HTML, malformed markup, mixed content).
- 62 existing tests plus 874 subtests all pass.
- **Benchmark**: heavy HTML with many `` tags went from 61.9ms to 1.1ms (98% faster). Math expressions got 44% faster.
- No regression on plain text or normal HTML.

### CODE REVIEW

File: src/yasbd/utils/cleaner.py [hunk 1]
```diff
@@ -53,14 +53,23 @@
...
 HTML_TAGS_FINDER = re.compile(
     r"""
-    # Branch 1: Strip the tag AND its content
-    <(script|img|iframe|object|embed|style|code)[^>]*?>.*?</\1>|
+    # HTML comments
+    <!--.*?-->|
 
-    # Branch 2: Just strip the brackets except quick formatting
-    </?\b[^libu][^>]*?>
+    # Declarations (<!DOCTYPE html>, <!ENTITY ...>, etc.)
```
[the remark this answers is not in the document]
Comment (speedyk-005) [reply src/yasbd/utils/cleaner.py:62]: <![^>]+> is a simple character class with no
 nested quantifiers. Literally impossible to cause
 catastrophic backtracking.
